tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java

```diff
@@ -0,0 +1,204 @@
+            .build())
+        .extension(new CassandraExtension())
+        .extension(new RabbitMQExtension())
+        .extension(new AwsS3BlobStoreExtension())
+        .server(configuration -> DistributedServer.createServer(configuration)
+            .overrideWith(new LinagoraTestJMAPServerModule())
+            .overrideWith(new JmapGuiceKeystoreManagerModule())
+            .overrideWith(binder -> Multibinder.newSetBinder(binder, GuiceProbe.class).addBinding().to(BlobStoreProbe.class)))
+        .build();
+
+    @AfterAll
+    static void afterAll() {
+        dockerAwsS3Container.stop();
+    }
+
+    @BeforeEach
+    void setUp(GuiceJamesServer server) throws Exception {
+        server.getProbe(DataProbeImpl.class)
+            .fluent()
+            .addDomain(DOMAIN.asString())
+            .addUser(BOB.asString(), BOB_PASSWORD)
+            .addUser(ANDRE.asString(), ANDRE_PASSWORD);
+
+        UserCredential userCredential = new UserCredential(BOB, BOB_PASSWORD);
+        PreemptiveBasicAuthScheme authScheme = new PreemptiveBasicAuthScheme();
+        authScheme.setUserName(userCredential.username().asString());
+        authScheme.setPassword(userCredential.password());
+        requestSpecification = new RequestSpecBuilder()
+            .setContentType(JSON)
+            .setAccept(JSON)
+            .setConfig(RestAssuredConfig.newConfig().encoderConfig(EncoderConfig.encoderConfig().defaultContentCharset(StandardCharsets.UTF_8)))
+            .setPort(server.getProbe(JmapGuiceProbe.class)
+                .getJmapPort()
+                .getValue())
+            .setBasePath(JMAPUrls.JMAP)
+            .setAuth(authScheme)
+            .addHeader(ACCEPT.toString(), ACCEPT_RFC8621_VERSION_HEADER)
+            .build();
+
+        MailboxProbeImpl mailboxProbe = server.getProbe(MailboxProbeImpl.class);
+        mailboxProbe.createMailbox(MailboxPath.inbox(BOB));
+        mailboxProbe.createMailbox(MailboxPath.inbox(ANDRE));
+
+        EncryptHelper.uploadPublicKey(ACCOUNT_ID, requestSpecification);
+    }
+
+    @Test
+    void sendEmailShouldResultingInSavingDataToBothObjectStorages(GuiceJamesServer server) {
+        given()
+            .body(LinagoraEmailSendMethodContract$.MODULE$.bobSendsAMailToAndre(server))
+        .when()
+            .post()
+        .then()
+            .statusCode(HttpStatus.SC_OK)
+            .contentType(JSON)
+            .extract()
+            .body()
+            .asString();
+
+        BlobStoreProbe blobStoreProbe = server.getProbe(BlobStoreProbe.class);
+        BucketName bucketName = Flux.from(blobStoreProbe.getFirstBlobStoreDAO().listBuckets()).collectList().block().getFirst();
+        List<BlobId> blobIds = Flux.from(blobStoreProbe.getFirstBlobStoreDAO().listBlobs(bucketName)).collectList().block();
+        List<BlobId> blobIds2 = Flux.from(blobStoreProbe.getSecondBlobStoreDAO().listBlobs(bucketName)).collectList().block();
+        assertThat(blobIds).isEqualTo(blobIds2);
```

Comment:
To be more robust we could try to ignore order (hasSameElementsAs + hasSameSizeAs) ?